src/aerialmap_display.cpp

```diff
@@ -466,56 +479,105 @@ void AerialMapDisplay::assembleScene()
   tileCache_.purge({ tileId, blocks_ });
 }
 
+namespace
+{
+// rotation matrix for rotating the current frame to ENU
+Ogre::Matrix3 rotationToENU(frame_convention convention)
+{
+  switch (convention)
+  {
+    case frame_convention::ned:
+    {
+      // XYZ->NED will cause the map to appear reversed when viewed from above (from +z).
+      // clang-format off
+      Ogre::Matrix3 const xyz_R_ned(0, 1, 0,
+                                    1, 0, 0,
+                                    0, 0, -1);
+      // clang-format on
+      return xyz_R_ned.Transpose();
+    }
+    case frame_convention::nwu:
+    {
+      // clang-format off
+      Ogre::Matrix3 const xyz_R_nwu(0, -1, 0,
+                                    1, 0, 0,
+                                    0, 0, 1);
+      // clang-format on
+      return xyz_R_nwu.Transpose();
+    }
+    case frame_convention::enu:
+    default:  // this should never be the case
+      return Ogre::Matrix3::IDENTITY;
+  }
+}
+}  // namespace
+
 void AerialMapDisplay::transformAerialMap()
 {
+  // We will use five frames in this function:
+  //
+  // * The frame from the NavSatFix message. It is rigidly attached to the robot.
+  // * Rviz's fixed frame (referred to as Ff). It is set by the user in Rviz.
+  // * The ENU/ NED/ NWU world frame "map". The code only uses ENU internally and rotates the frame into NED or NWU at
+  //   the end of the calculations.
+  // * The frame of the tiles. We assume that the tiles are in a frame where x points eastwards and y southwards. This
+  //   frame is used by OSM and Google Maps, see https://developers.google.com/maps/documentation/javascript/coordinates
+  // * The AerialMap frame is an ENU frame. This is where all tiles are placed in. The center tile is in the first
+  //   quadrant, to be exact the left bottom corner of the center tile has coordinates (0,0) in the AerialMap frame.
+
   if (!isEnabled())
   {
     return;
   }
 
-  // the parent frame of this scene node
-  static std::string const mapFrame = "map";
-
-  // the robot's frame
-  std::string const frame = ref_fix_.header.frame_id;
+  // the frame "map" is defined here: https://www.ros.org/reps/rep-0105.html#map
+  std::string const static frameMap = "map";
+  std::string const frameNavSatFix = ref_fix_.header.frame_id;
 
-  // note: the orientation is not used on purpose
-  Ogre::Quaternion orientation;
-  Ogre::Vector3 position;
+  // unused variables
+  Ogre::Vector3 translationUnused;
+  Ogre::Quaternion rotationUnused;
 
-  // can mapFrame be used as a fixed frame?
-  std::string errMsg;
-  if (context_->getFrameManager()->frameHasProblems(mapFrame, ros::Time{}, errMsg) ||
-      context_->getFrameManager()->transformHasProblems(mapFrame, ros::Time{}, errMsg))
+  // rotation from map frame into the fixed frame
+  Ogre::Quaternion rotationFfToMap;
```

Comment:
FfToMap or MapToFf - either doc or naming is twisted

Reply:
Good catch!